feat: Store txHash and blockHash in private event storage

mverzilli opened the pull request:
First part of F-96 (about returning event metadata)

We want the PXE to expose some event metadata (mainly tx hash, block number, and block hash) to wallets.

I plan to tackle this in little self contained PR's. This first PR only affects internals, but not APIs: it modifies the PXE's private event store to store tx hash and block hash for later consumption.

I'm introducing a `PrivateEvent` type which for now is an implementation detail as it is swallowed by PXE. But eventually it could be lifted to the PXE API. 

I'm not yet familliarized enough with criteria for types to be defined and made available by stdlib or Aztec.js, it might make sense to eventually have a re-usable `PrivateEvent` type _somewhere over there_, but one thing at a time.

There's also a very mildly annoying amount of duplication in `private_event_data_provider.test.ts` which I'm contributing to with this PR, but I'm opting to tackle that in subsequent PRs as I work on a more mature API.

benesjan reviewed (approved):
Very nice 👍

nventuro commented on yarn-project/pxe/src/storage/private_event_data_provider/private_event_data_provider.ts:
```diff
@@ -4,14 +4,25 @@ import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
 
+export type PrivateEvent = {
+  msgContent: Fr[];
+  blockNumber: number;
+  blockHash: L2BlockHash;
+  txHash: TxHash;
```
There's an `InBlock` wrapper that could be used here. Not saying _we have_ to use it, but other people have found it useful in similar contexts. Up to you.

nventuro reviewed (approved):
Great, thanks!

Don't be afraid of opening larger PRs if you feel like it'd be faster/easier for you. My rule of thumb for a too-large PR is around 500 lines.

mverzilli committed:
feat: Store txHash and blockHash in private event storage

First part of F-96 (about returning event metadata)

We want the PXE to expose some event metadata (mainly tx hash, block number, and block hash) to wallets.

I plan to tackle this in little self contained PR's. This first PR only affects internals, but not APIs: it modifies the PXE's private event store to store tx hash and block hash for later consumption.

I'm introducing a `PrivateEvent` type which for now is an implementation detail as it is swallowed by PXE. But eventually it could be lifted to the PXE API.

I'm not yet familliarized enough with criteria for types to be defined and made available by stdlib or Aztec.js, it might make sense to eventually have a re-usable `PrivateEvent` type _somewhere over there_, but one thing at a time.

There's also a very mildly annoying amount of duplication in `private_event_data_provider.test.ts` which I'm contributing to with this PR, but I'm opting to tackle that in subsequent PRs as I work on a more mature API.
